tasks/check-data-keys/check-data-keys.yaml

```
@@ -24,52 +29,52 @@ spec:
       description: The workspace where the data JSON file resides
   steps:
     - name: check-data-keys
-      image: quay.io/konflux-ci/release-service-utils:e633d51cd41d73e4b3310face21bb980af7a662f
+      # TODO: Update the image to the latest release-service-utils image
+      image: quay.io/redhat-user-workloads/rhtap-release-2-tenant/release-service/release-service-utils:on-pr-403edb0827b86a46933608edc09a8b8598373f0a 
       args: ["$(params.systems[*])"]
       script: |
         #!/usr/bin/env bash
         set -ex
 
-        KEYS_JSON='{
-            "releaseNotes": [
-                "product_id",
-                "product_name",
-                "product_version",
-                "product_stream",
-                "cpe",
-                "type",
-                "content.images",
-                "synopsis",
-                "topic",
-                "description",
-                "solution",
-                "references"
-            ],
-            "cdn": [
-                "env"
-            ]
-        }'
+        if [ ! -f "/workspace/data/$(params.dataPath)" ]; then
+            echo "Data JSON file not found: /workspace/data/$(params.dataPath)"
+            exit 1
+        fi
+
+        if ! curl -s "$(params.schema)" -o /tmp/schema.json; then
```

By default, curl will return 0 even if the response has code >400:
```
$curl -s https://raw.githubusercontent.com/seanconroy2021/release-service-catalog/refs/heads/RELEASE-987/bla
404: Not Found%
$ echo $?
0
```
So you'll need `--fail` or `--fail-with-body` - probably the latter, otherwise you will not know what failed.

So you'll have:
```suggestion
        if ! curl -s --fail-with-body "$(params.schema)" -o /tmp/output; then
          echo "Error: Failed to fetch schema from $(params.schema)"
          cat /tmp/output
          exit
        fi
```

---

I will add it in. Ty

---

With the new approach, we can now pass in the schema URL to the validator, which will handle the error if the JSON can't be found. 
` check-jsonschema --output-format=text --schemafile "$(params.schema)" "$(workspaces.data.path)/$(params.dataPath)"`